kinds/dashboard/dashboard_kind.cue

```diff
@@ -137,12 +138,161 @@ lineage: seqs: [
 					state:         #LoadingState
 					error?: {...}
 					description?: string
-					// TODO: Move this into a separated QueryVariableModel type
-					query?:      string | {...}
+				} @cuetsy(kind="interface")
+
+				// Variables which allow to select filters from a datasource based on the dimensions from the datasource.
+				#AdHocVariableModel: {
+					#BaseVariableModel
+					type: "adhoc"
+					// Datasource ref can be defined and hold null values
 					datasource?: #DataSourceRef
-					...
-				} @cuetsy(kind="interface") @grafana(TSVeneer="type") @grafanamaturity(NeedsExpertReview)
-				#VariableHide: 0 | 1 | 2                                                 @cuetsy(kind="enum",memberNames="dontHide|hideLabel|hideVariable") @grafana(TSVeneer="type") @grafanamaturity(NeedsExpertReview)
+					filters: [...#AdHocVariableFilter]
+				} @cuetsy(kind="interface") @grafana(TSVeneer="type")
+
+				// Filters selected filters generated for an ad-hoc variable from a datasource.
+				#AdHocVariableFilter: {
+					key:       string
+					operator:  string
+					value:     string
+					condition: string
+				} @cuetsy(kind="interface")
+
+				// Common props for variables injected by the system.
+				#SystemVariable: {
+					#BaseVariableModel
+					type:        "system"
+					systemValue: string
+					current: {
+						value: {...}
+					}
+				} @cuetsy(kind="interface")
+
+				// Variable injected by the system which holds the current dashboard.
+				#DashboardVariableModel: {
```

Comment:
```suggestion
				#DashVariableModel: {
```

Reply:
so if we do this, it should at least make the Go types compile, because we won't have an issue with the prefix dropper getting rid of `Dashboard` anymore.

If you do this, then you can hackily re-export it from the [common veneer](https://github.com/grafana/grafana/blob/3006a457f2ca62369149fb24de02c757bb6bec9a/packages/grafana-schema/src/veneer/common.types.ts#L1). Then, when we're able to move these out of the dashboard kind and into the commons, there won't be an issue with package name conflicts anymore.

Reply:
Are we planning to solve this name dropping? Maybe we can avoid it with a custom tag. It will create inconsistencies if we use veneers for renaming. Anyways, I think it is good for now :)

Reply:
> Are we planning to solve this name dropping? Maybe we can avoid it with a custom tag. It will create inconsistencies if we use veneers for renaming. Anyways, I think it is good for now :)

We cannot use Veneers for renaming, because when regenerating `index.gen.ts`, the export is generated with the schema name (it makes total sense). How can we solve this without exporting "Dash" prefixed variables?

Reply:
> We cannot use Veneers for renaming, because when regenerating index.gen.ts, the export is generated with the schema name (it makes total sense). 

What if you drop `@grafana(TSVeneer="type")` for this variable and add the veneer manually? So in `dashboard.types` you do `import { DashVariableModel as DashboardVariableModel }` and re-export it, could it work?